src/in-mem/backend.service.ts

```diff
@@ -160,7 +160,7 @@ export abstract class BackendService {
    */
   protected addDelay(response: Observable<any>): Observable<any> {
     const d = this.config.delay;
-    return d === 0 ? response : delay.call(response, d || 500);
```

Comment:
Replaces RxJS `delay()` (incorrect for errors) with new `delayRespons()` function.